src/diart/models.py

```diff
@@ -1,28 +1,41 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Text, Union, Callable
+from typing import Optional, Text, Union, Callable, Mapping, TYPE_CHECKING
 
 import torch
 import torch.nn as nn
+from requests import HTTPError
 
 try:
     import pyannote.audio.pipelines.utils as pyannote_loader
+    from pyannote.audio import Inference, Model
+    from pyannote.audio.pipelines.speaker_verification import (
+        WeSpeakerPretrainedSpeakerEmbedding,
+    )
 
     _has_pyannote = True
 except ImportError:
     _has_pyannote = False
 
+if TYPE_CHECKING:
+    from pyannote.audio.pipelines.speaker_verification import (
+        WeSpeakerPretrainedSpeakerEmbedding,
+    )
+
 
 class PyannoteLoader:
     def __init__(self, model_info, hf_token: Union[Text, bool, None] = True):
         super().__init__()
         self.model_info = model_info
         self.hf_token = hf_token
 
-    def __call__(self) -> nn.Module:
-        return pyannote_loader.get_model(self.model_info, self.hf_token)
+    def __call__(self) -> Union[nn.Module, WeSpeakerPretrainedSpeakerEmbedding]:
+        try:
+            return pyannote_loader.get_model(self.model_info)
+        except HTTPError:
+            return WeSpeakerPretrainedSpeakerEmbedding(self.model_info)
```

Comment:
I like this idea of first assuming we were given a model, then if that fails let's assume it's an embedding pipeline. However, I would like this to be more generic than just for wespeaker embeddings.

I suggest the following:
```suggestion
    def __call__(self) -> Callable:
        try:
            return Model.from_pretrained(self.model_info, use_auth_token=self.hf_token)
        except HTTPError:
            return PretrainedSpeakerEmbedding(self.model_info, use_auth_token=self.hf_token)
```

Here we replace the old `get_model` with a cleaner `Model.from_pretrained`, and if that fails we try to find the corresponding pre-trained speaker embedding pipeline. It leaves space for both segmentation and embedding models of any form, and the return type is generic enough to encompass all of them.

Reply:
the return type could also be `Union[Model, PretrainedSpeakerEmbedding]` if we want to be precise

Reply:
That's very smooth. Added it.